src/main/java/com/manywho/services/saml/adapters/ManyWhoSaml2Settings.java

```diff
@@ -67,4 +81,16 @@ public static X509Certificate parseCertificate(String certificate) throws Certif
 
         return (X509Certificate) CertificateFactory.getInstance("X.509").generateCertificate(is);
     }
+
+    public static PrivateKey parsePrivateKey(String privateKey) throws Base64DecodingException, InvalidKeySpecException, NoSuchAlgorithmException {
+        String privateKeyBody = privateKey
+            .replaceAll("-----BEGIN PRIVATE KEY-----", "")
+            .replaceAll("-----END PRIVATE KEY-----", "");
+
+        byte[] decoded = java.util.Base64.getDecoder().decode(privateKeyBody);
+
+        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+        return keyFactory.generatePrivate(new PKCS8EncodedKeySpec(decoded));
```

Comment:
@matt-manywho I haven't found a better way to dynamically detect from the private key if it is actually an RSA.
But I think we should throw an exception that explains that only RSA is currently supported in case of generatePrivate key throws an Exception.
````
throw new RuntimeException("There was an error decoding the Private Key (only RSA is currently supported"), exception)"
````